hydromt_wflow/components/staticmaps.py

```diff
@@ -1,11 +1,254 @@
+
+        Parameters
+        ----------
+        filename : str, optional
+            filename relative to model root, by default 'grid/grid.nc'
+        mask_and_scale : bool, optional
+            If True, replace array values equal to _FillValue with NA and scale values
+            according to the formula original_values * scale_factor + add_offset, where
+            _FillValue, scale_factor and add_offset are taken from variable attributes
+            (if they exist).
+        **kwargs : dict
+            Additional keyword arguments to be passed to the `read_nc` method.
+        """
+        self.root._assert_read_mode()
+        self._initialize_grid(skip_read=True)
+
+        # Sort which path/ filename is actually the one used
+        # Hierarchy is: 1: signature, 2: config, 3: default
+        p = filename or self.model.config.get("input.path_static") or self._filename
+        # Check for input dir
+        p_input = self.model.config.get("dir_input", fallback="")
+
+        # Load grid data in r+ mode to allow overwriting netcdf files
+        if self.root.is_reading_mode() and self.root.is_writing_mode():
+            kwargs["load"] = True
+        loaded_nc_files = _read_nc(
+            Path(p_input, p),
+            self.root.path,
+            single_var_as_array=False,
+            mask_and_scale=mask_and_scale,
+            **kwargs,
+        )
+        for ds in loaded_nc_files.values():
+            if ds.raster.res[1] > 0:  # Make sure its north-south oriented
+                ds = ds.raster.flipud()
+            self.set(ds)
+
+    @hydromt_step
+    def write(
+        self,
+        filename: Path | str | None = None,
+        **kwargs,
+    ):
+        """Write staticmaps model data.
+
+        Key-word arguments are passed to :py:meth:`~hydromt.model.Model.write_nc`
+
+        Parameters
+        ----------
+        filename : str, optional
+            filename relative to model root, by default None
+        **kwargs : dict
+            Additional keyword arguments to be passed to the `write_nc` method.
+        """
+        # Solve pathing same as read
+        # Hierarchy is: 1: signature, 2: config, 3: default
+        p = filename or self.model.config.get("input.path_static") or self._filename
+        # Check for input dir
+        p_input = Path(self.model.config.get("dir_input", fallback=""), p)
+
+        # Supercharge with the base grid component write method
+        super().write(
+            p_input.as_posix(),
+            gdal_compliant=True,
+            rename_dims=True,
+            **kwargs,
+        )
+
+        # Set the config entry to the correct path
+        self.model.config.set(
+            "input.path_static",
+            Path(self.root.path, p_input).as_posix(),
+        )
+
+    ## Mutating methods
+    def set(
+        self,
+        data: xr.DataArray | xr.Dataset,
+        name: str | None = None,
+        mask: str | xr.DataArray | None = None,
```

Comment:
I don't think this needs to be an argument? It's only temporay that we still have model.set_grid to avoir merge conflicts but it will be removed at some point (especially since it is not even an hydromt_step) so you can fully transfer now already.